tests/test_public_api.py

```diff
@@ -14,6 +14,58 @@ def model_setup(backend):
     return model, data, init_pars
 
 
+def test_set_backend_by_name(backend):
+    tb, _ = pyhf.get_backend()
+    pyhf.set_backend(tb.name)
```

Comment:
this is kinda confusing, since `set_backend()` is called with the `backend` fixture